diffsync/store/redis.py

```diff
@@ -0,0 +1,242 @@
+        """Init method for RedisStore."""
+        super().__init__(*args, **kwargs)
+
+        try:
+            if url:
+                self._store = Redis.from_url(url, db=db)
+            else:
+                self._store = Redis(host=host, port=port, db=db)
+
+            if not self._store.ping():
+                raise RedisConnectionError
+        except RedisConnectionError:
+            raise ObjectStoreException("Redis store is unavailable.") from RedisConnectionError
+
+        self._store_id = store_id if store_id else str(uuid.uuid4())[:8]
+
+        self._store_label = f"{REDIS_DIFFSYNC_ROOT_LABEL}:{self._store_id}"
+
+    def __str__(self):
+        """Render store name."""
+        return f"{self.name}({self._store_id})"
+
+    def get_all_model_names(self):
+        """Get all the model names stored.
+
+        Return:
+            List[str]: List of all the model names.
+        """
+        # TODO: optimize it
+        all_model_names = []
+        for item in self._store.scan_iter(f"{self._store_label}:*"):
+            model_name = item.split(b":")[2].decode()
+            if model_name not in all_model_names:
+                all_model_names.append(model_name)
+
+        return all_model_names
+
+    def _get_key_for_object(self, modelname, uid):
+        return f"{self._store_label}:{modelname}:{uid}"
+
+    def get(self, obj: Union[Text, "DiffSyncModel", Type["DiffSyncModel"]], identifier: Union[Text, Mapping]):
+        """Get one object from the data store based on its unique id.
+
+        Args:
+            obj: DiffSyncModel class or instance, or modelname string, that defines the type of the object to retrieve
+            identifier: Unique ID of the object to retrieve, or dict of unique identifier keys/values
+
+        Raises:
+            ValueError: if obj is a str and identifier is a dict (can't convert dict into a uid str without a model class)
+            ObjectNotFound: if the requested object is not present
+        """
+        if isinstance(obj, str):
+            modelname = obj
+            if not hasattr(self, obj):
+                object_class = None
+            else:
+                object_class = getattr(self, obj)
+        else:
+            object_class = obj
+            modelname = obj.get_type()
+
+        if isinstance(identifier, str):
+            uid = identifier
+        elif object_class:
+            uid = object_class.create_unique_id(**identifier)
+        else:
+            raise ValueError(
+                f"Invalid args: ({obj}, {identifier}): "
+                f"either {obj} should be a class/instance or {identifier} should be a str"
+            )
+
+        try:
+            obj_result = loads(self._store.get(self._get_key_for_object(modelname, uid)))  # nosec
+            obj_result.diffsync = self.diffsync
+        except TypeError:
+            raise ObjectNotFound(f"{modelname} {uid} not present in Cache")  # pylint: disable=raise-missing-from
```

Comment:
Why `disable=raise-missing-from` instead of just using `from exc` here and below?